src/Public/New-CCMDeploymentStep.ps1

```diff
@@ -144,18 +154,20 @@ function New-CCMDeploymentStep {
         switch ($PSCmdlet.ParameterSetName) {
             'Basic' {
                 $Body = @{
-                    Name                         = "$Name"
-                    DeploymentPlanId             = "$(Get-CCMDeployment -Name $Deployment | Select-Object -ExpandProperty Id)"
-                    DeploymentStepGroups         = @(
-                        Get-CCMGroup -Group $TargetGroup | Select-Object Name, Id | ForEach-Object {
-                            [pscustomobject]@{ groupId = $_.id; groupName = $_.name }
-                        }
+                    name                           = $Name
+                    deploymentPlanId               = (Get-CCMDeployment -Name $Deployment).id
+                    deploymentStepGroups           = @(

```

Comment:
```suggestion
                    deploymentStepGroups           = [System.Collections.Generic.List[psobject]]@(
```

This cast ensures the collection is always preserved during json serialization; it seems some versions of powershell unwrap a one-element array, but leave other collection types intact.